qiskit/circuit/library/standard_gates/x.py

```diff
@@ -962,37 +972,17 @@ def __init__(
 
     def inverse(self):
         """Invert this gate. The MCX is its own inverse."""
-        return MCXGate(num_ctrl_qubits=self.num_ctrl_qubits, ctrl_state=self.ctrl_state)
-
-    @staticmethod
-    def get_num_ancilla_qubits(num_ctrl_qubits: int, mode: str = "noancilla") -> int:
-        """Get the number of required ancilla qubits without instantiating the class.
-
-        This staticmethod might be necessary to check the number of ancillas before
-        creating the gate, or to use the number of ancillas in the initialization.
-        """
-        if mode == "noancilla":
-            return 0
-        if mode in ["recursion", "advanced"]:
-            return int(num_ctrl_qubits > 4)
-        if mode[:7] == "v-chain" or mode[:5] == "basic":
-            return max(0, num_ctrl_qubits - 2)
-        raise AttributeError(f"Unsupported mode ({mode}) specified!")
+        return MCXGate(num_ctrl_qubits=self.num_ctrl_qubits, ctrl_state=self.ctrl_state,
+                       _name=self.name, synthesis=self.synthesis)
 
     def _define(self):
         """The standard definition used the Gray code implementation."""
-        # pylint: disable=cyclic-import
-        from qiskit.circuit.quantumcircuit import QuantumCircuit
-
-        q = QuantumRegister(self.num_qubits, name="q")
-        qc = QuantumCircuit(q)
-        qc._append(MCXGrayCode(self.num_ctrl_qubits), q[:], [])
-        self.definition = qc
+        self.definition = self.synthesis.synthesize(self.num_ctrl_qubits)
```

Comment:
Could we pass the entire gate into the synthesis algorithm, i.e. `self.definition = self.synthesis.synthesize(self)`? This is what we adopted in the `PauliEvolutionGate` and that is likely more future-proof since the gate can contain more information than just the number of control qubits.

Reply:
Oh, that's really cool and makes a lot of sense. A follow-up question for `MCXSynthesisVChain`: does it make more sense for `dirty_ancillas` to be a part of the gate information -- thus allowing `self.synthesis.synthesize(self)` to check `self.dirty_ancillas`, or does it make more sense as it is now -- with `dirty_ancillas` being passed in `MCXSynthesisVChain.__init__`?

Reply:
I think `dirty_ancillas` is good where it is now: in the synthesis. Since that's an information on how the gate will be compiled, it should not be in the `MCXGate` but rather in the synthesis, so users can invoke it something like
```python
mcx = MCXGate(.., synthesis=VChain(dirty_ancillas=True)
```